plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,284 @@
+    maybeAdd(violations, validatePortNotInUse(pathOf("httpsPort"), configToValidate.httpsPort));
+    maybeAdd(violations, validateNotSamePort(
+        pathOf("httpPort"), configToValidate.httpPort, pathOf("httpsPort"), configToValidate.httpsPort
+    ));
+    ConstraintViolations.maybePropagate(violations, log);
+  }
+
+  @Override
+  protected void doConfigure(final Configuration configuration) throws Exception {
+    config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    log.debug("Config: {}", config);
+  }
+
+  @Override
+  protected void doStart() throws Exception {
+    if (config.httpPort != null) {
+      try {
+        httpConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTP, config.httpPort);
+        connectorRegistrar.addConnector(httpConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTP connector on port {} for docker repository {}",
+            config.httpPort, getRepository().getName(), e
+        );
+      }
+    }
+    if (config.httpsPort != null) {
+      try {
+        httpsConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTPS, config.httpsPort);
+        connectorRegistrar.addConnector(httpsConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTPS connector on port {} for docker repository {}",
+            config.httpsPort, getRepository().getName(), e
+        );
+      }
+    }
+  }
+
+  @Override
+  protected void doStop() throws Exception {
+    if (httpConfig != null) {
+      connectorRegistrar.removeConnector(httpConfig);
+    }
+    if (httpsConfig != null) {
+      connectorRegistrar.removeConnector(httpsConfig);
+    }
+  }
+
+  @Override
+  public Integer getHttpPort() {
+    return config.httpPort;
+  }
+
+  @Override
+  public Integer getHttpsPort() {
+    return config.httpsPort;
+  }
+
+  /**
+   * Validate that port is not used as used as HTTP/HTTPS ports in another repository.
+   */
+  private ConstraintViolation<?> validatePortNotInUse(final String path, final Integer port) {
+    if (port != null) {
+      Iterable<Repository> repositories;
+      try {
+        repositories = repositoryManager.browse();
+      }
+      catch (IllegalStateException ignored) { //NOSONAR
+        // this will happen when facet is validated during startup of repository manager (load from storage)
+        // ignore and consider configurations loaded from storage as valid
+        repositories = Collections.emptyList();
+      }
+      for (Repository repository : repositories) {
+        if (DockerFormat.NAME.equals(repository.getFormat().getValue()) && !repository.equals(getRepository())) {
+          DockerConnectorFacet connectorFacet = repository.facet(DockerConnectorFacet.class);
+          if (port.equals(connectorFacet.getHttpPort()) || port.equals(connectorFacet.getHttpsPort())) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)